app/src/androidTest/java/com/example/antitheft/structure/LocationParserTest.java

```diff
@@ -0,0 +1,179 @@
+package com.example.antitheft.structure;
+
+import android.location.Location;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test of {@link LocationParser} class.
+ *
+ * @author Francesco Bau'
+ * @see LocationParser for more info about the tested class.
+ * @since 28/02/2020
+ */
+public class LocationParserTest {
+
+    private static final float NOT_DEFAULT_ACCURACY = Float.MAX_VALUE;
+    LocationParser locationParser;
+
+    double expectedLatitude = Double.MAX_VALUE;
+    double expectedLongitude = Double.MIN_VALUE;
+
+    /**
+     * Before starting the tests, the {@link LocationParser} instance needs to be initialized.
+     *
+     * @see Location
+     * @see LocationParser
+     */
+    @Before
+    public void init() {
+        locationParser = new LocationParser();
+        Location location = new Location(LocationParser.DEFAULT_PROVIDER);
+        location.setLatitude(expectedLatitude);
+        location.setLongitude(expectedLongitude);
+        locationParser.setLocation(location);
+    }
+
+    /**
+     * Testing the method {@link LocationParser#getLatitude()}.
+     *
+     * @see LocationParser
+     * @see LocationParser#getLatitude()
+     */
+    @Test
+    public void testLatitude() {
+        assertEquals(true, expectedLatitude == locationParser.getLatitude());
```

Comment:
Actually not, because locationParser.getLatitude() returns a double... locationParser is a LocationParser variable, and LocationParser class is made by me...

Here's a link to the class LocationParser:
https://github.com/francescobau/AntiTheft/blob/feature-locationCommand-CR/app/src/main/java/com/example/antitheft/structure/LocationParser.java

Reply:
Also, assertEquals(double,double) is deprecated, that's why I was forced to do it this way...

Reply:
You are right. You could write `assertTrue(expectedLatitude == locationParser.getLatitude());` or leave it as it is.